va/va.go

```diff
@@ -486,80 +523,117 @@ func (va *ValidationAuthorityImpl) performRemoteValidation(
-			bad++
+		} else if resp.result.Problems != nil {
+			// The remote VA returned a problem.
+			failed = append(failed, resp.rva)
 
 			var err error
-			currProb, err = bgrpc.PBToProblemDetails(res.response.Problems)
+			currProb, err = bgrpc.PBToProblemDetails(resp.result.Problems)
 			if err != nil {
-				va.log.Errf("Remote VA %q.PerformValidation returned malformed problem: %s", res.hostname, err)
+				va.log.Errf("Remote VA %q.PerformValidation returned malformed problem: %s", perspective, err)
 				currProb = probs.ServerInternal("Remote PerformValidation RPC returned malformed result")
 			}
 		} else {
-			good++
+			// The remote VA returned a successful result.
+			passed = append(passed, resp.rva)
 		}
 
 		if firstProb == nil && currProb != nil {
 			firstProb = currProb
 		}
 
-		// Return as soon as we have enough successes or failures for a definitive result.
-		if good >= required {
-			return nil
+		if len(passed) >= required {
+			// Enough successful responses to reach quorum.
+			return va.summarizeMPIC(passed, failed), nil
 		}
-		if bad > va.maxRemoteFailures {
+		if len(failed) > va.maxRemoteFailures {
+			// Too many failed responses to reach quorum.
 			va.metrics.remoteValidationFailures.Inc()
 			firstProb.Detail = fmt.Sprintf("During secondary validation: %s", firstProb.Detail)
-			return firstProb
+			return va.summarizeMPIC(passed, failed), firstProb
 		}
 
 		// If we somehow haven't returned early, we need to break the loop once all
 		// of the VAs have returned a result.
-		if good+bad >= len(va.remoteVAs) {
+		if len(passed)+len(failed) >= remoteVACount {
 			break
 		}
 	}
 
-	// This condition should not occur - it indicates the good/bad counts neither
-	// met the required threshold nor the maxRemoteFailures threshold.
-	return probs.ServerInternal("Too few remote PerformValidation RPC results")
+	// This condition should not occur - it indicates the passed/failed counts
+	// neither met the required threshold nor the maxRemoteFailures threshold.
+	return va.summarizeMPIC(passed, failed), probs.ServerInternal("Too few remote PerformValidation RPC results")
+}
+
+func (va *ValidationAuthorityImpl) summarizeMPIC(passed, failed []RemoteVA) *mpicSummary {
+	var passedPerspectives []string
+	var failedPerspectives []string
+	var passedRIRs []string
+	for _, rva := range passed {
+		passedPerspectives = append(passedPerspectives, rva.Perspective)
+		passedRIRs = append(passedRIRs, rva.RIR)
+	}
+	for _, rva := range failed {
+		failedPerspectives = append(failedPerspectives, rva.Perspective)
+		passedRIRs = append(passedRIRs, rva.RIR)
+	}
+
+	if passedPerspectives == nil {
+		passedPerspectives = []string{}
+	}
+	if failedPerspectives == nil {
+		failedPerspectives = []string{}
+	}
+	if passedRIRs == nil {
+		passedRIRs = []string{}
+	}
+
+	return &mpicSummary{
+		PassedPerspectives: passedPerspectives,
+		FailedPerspectives: failedPerspectives,
+		PassedRIRs:         passedRIRs,
```

Comment:
We're collecting up the RIRs but we're not ensuring that passing validations came from at least two unique RIRs.

Reply:
This is true. My goal here was to add logging, not enforcement.